scripts/APPS-162-delete-emails.js

```diff
@@ -0,0 +1,315 @@
+/*==========================
+ * APPS-162
+ *
+ * @description: Script to delete subscriber e-mails from a given topic
+ * @author: Government of Canada; Josh Berard and Jonas Graham
+ * @version: 1.0
+ ===========================*/
+
+
+const topicIds = ["hc-covid19en", "hc-covid19fr"]; // Replace with an array of topicIds for which you wish to clear e-mail addresses
+
+const dotenv = require('dotenv'); // Application configuration
+const path = require('path');
+const chalk = require('chalk'); // To color message in console log
+const MongoClient = require('mongodb').MongoClient;
+
+/**
+ * Load environment variables from .env file, where API keys and passwords are configured.
+ */
+dotenv.config({
+    path: '.env'
+});
+
+const processEnv = process.env;
+let db;
+
+MongoClient.connect( processEnv.MONGODB_URI || '', {useUnifiedTopology: true} ).then( ( mongoInstance ) => {
+
+		db = mongoInstance.db( processEnv.MONGODB_NAME || 'subs' );
+		
+		run();
+
+	}).catch( (e) => { console.log( "%s MongoDB ERRROR: %s", chalk.red('✗'), e ) } );
+
+
+async function run() {
+	let topicIdsString = topicIds.join(', ');
+	console.log(`Deleting e-mail addresses associated with ${topicIdsString} ...`);
+	await clearFromLogs();
+	let collectionsToClearEmailsFrom = ["subsConfirmed", "subsConfirmedNewCode", "subsUnconfirmed", "subsExist", "subsRecents", "subs_Unsubs"];
+	
+	for (let coll of collectionsToClearEmailsFrom) {
+		let emailCount = await countEmailsForTopicIdByCollection(topicIds, coll);
+		console.log(`found ${emailCount} entries for the specified topicIds in ${coll}`);
+	}
+	
+	for (let coll of collectionsToClearEmailsFrom) {
+		let emailCount = await deleteEmailsForTopicIdByCollection(topicIds, coll);
+		console.log(`Deleted entries for the specified topicIds in ${coll}`);
+	}
+	
+	for (let coll of collectionsToClearEmailsFrom) {
+		let emailCount = await countEmailsForTopicIdByCollection(topicIds, coll);
+		console.log(`found ${emailCount} entries for the specified topicIds in ${coll}`);
+	}
+	
+	let subsExistEmailCount = await db.collection("subsExist").countDocuments({t: {$in: topicIds}});
+	console.log(`found ${subsExistEmailCount} entries for the specified topicIds in subsExist`);
+	await db.collection("subsExist").deleteMany({t: {$in: topicIds}}); 
+	console.log(`Deleted entries for the specified topicIds in subsExist`);
+	subsExistEmailCount = await db.collection("subsExist").countDocuments({t: {$in: topicIds}});
+	console.log(`found ${subsExistEmailCount} entries for the specified topicIds in subsExist`);
+	
+	console.log(chalk.green("✓ DONE!"));	
```

Comment:
Missing the removal of the topic from the collection "topic" and "topics_details". Or at least to invalidate its notifyKey + confirmURL + unsubURL + failURL in the "topic" collection to ensure that topic is deactivated and out of order.

Reply:
Hmmm...  The request was just to delete the e-mail addresses, not remove any trace of the topic, entirely.  Perhaps they might wanna keep _some_ record of the topic having existed?

Reply:
I went with your second suggestion for now.  See the _invalidateTopics_ function.